renderer_test.go

```diff
@@ -0,0 +1,190 @@
+package atlassianwikirenderer
+
+import (
+	"bytes"
+	"strings"
+	"testing"
+
+	"github.com/yuin/goldmark"
+)
+
+type testCase struct {
+	input    string
+	expected string
+}
+
+func TestTaskCheckbox(t *testing.T) {
+	runTest(t, testCase{
+		input:    "- [ ] Task 1\n- [x] Task 2 (completed)\n- [ ] Task 3",
+		expected: "*  [ ] Task 1\n*  [x] Task 2 (completed)\n*  [ ] Task 3",
+	})
+}
+
+func TestHeadingRendering(t *testing.T) {
+	runTest(t, testCase{
+		input:    "# Heading 1",
+		expected: "h1.Heading 1",
+	})
+}
+
+func TestBlockquoteRendering(t *testing.T) {
+	runTest(t, testCase{
+		input:    "> This is a blockquote",
+		expected: "{quote}\nThis is a blockquote\n{quote}",
+	})
+}
+
+func TestFencedCodeBlockRendering(t *testing.T) {
+	runTest(t, testCase{
+		input:    "```This is a fenced code block```",
+		expected: "{{This is a fenced code block}}",
+	})
+}
+
+func TestParagraphRendering(t *testing.T) {
+	runTest(t, testCase{
+		input:    "This is a paragraph. It can contain text,\nline breaks, and various formatting options.\n\nYou can separate paragraphs by leaving a blank line between them.",
+		expected: "This is a paragraph. It can contain text,\n\nline breaks, and various formatting options.\n\nYou can separate paragraphs by leaving a blank line between them.",
+	})
+}
+
+func TestCodeBlockRendering(t *testing.T) {
+	runTest(t, testCase{
+		input:    "\t`Code Block`",
+		expected: "{code}{code}",
+	})
+}
+
+func TestCodeSpanRendering(t *testing.T) {
+	runTest(t, testCase{
+		input:    "This is an `inline code span`",
+		expected: "This is an {{inline code span}}",
+	})
+}
+
+func TestEmphasisRendering(t *testing.T) {
+	runTest(t, testCase{
+		input:    "*Emphasised text*",
+		expected: "_Emphasised text_",
+	})
+}
+
+func TestLinkRendering(t *testing.T) {
+	runTest(t, testCase{
+		input:    "[Link](https://www.jfrog.com)",
+		expected: "[Link|https://www.jfrog.com]",
+	})
+}
+
+func TestAutoLinkRendering(t *testing.T) {
+	runTest(t, testCase{
+		input:    "<https://www.jfrog.com>",
+		expected: "[https://www.jfrog.com]",
+	})
+}
+
+func TestStrikethroughRendering(t *testing.T) {
+	runTest(t, testCase{
+		input:    "~~Strikethrough~~",
+		expected: "~~Strikethrough~~",
```

Comment:
This is a bug, the correct expectation is, will add a separate ticket to track this.

```
func TestStrikethroughRendering(t *testing.T) {
	runTest(t, testCase{
		input:    "~~Strikethrough~~",
		expected: "-Strikethrough-",
	})
}
```